### ISSUE TITLE
Snapshot name causes rate limit errors?

### ISSUE THREAD

**Issue by aljazsipec** [sabeechen/hassio-google-drive-backup#299]
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

 Addon version: `0.101.3`
 Home Assistant Version: `0.117.1`
 Supervisor Version: `2020.10.0`
 Supervisor Channel: `stable`
 Hassos Version: `4.15`
 Docker Version: `19.03.11`
 Architecture: `aarch64`
 Machine: `raspberrypi4-64`
 Date: `2020-11-01T17:12:54.464245+00:00`
 Timezone: `Europe/Ljubljana`
 Failure Time: `2020-11-01T17:10:53.399193+00:00`
 Last Good Sync: `2020-11-01T17:12:26.010917+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:162 (sync)
  addon/backup/drive/drivesource.py:150 (save)
  addon/backup/drive/driverequests.py:201 (create)
  addon/backup/drive/driverequests.py:342 (retryRequest)
  addon/backup/util/backoff.py:36 (backoff)
  addon/backup/drive/driverequests.py:330 (retryRequest)
  addon/backup/creds/driverequester.py:38 (request)
backup.exceptions.exceptions.GoogleRateLimitError

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 2,
        "retained": 0,
        "deletable": 2,
        "name": "HomeAssistant",
        "latest": "2020-11-01T14:04:23Z",
        "size": "124.6 MB"
    },
    "GoogleDrive": {
        "snapshots": 0,
        "retained": 0,
        "deletable": 0,
        "name": "GoogleDrive",
        "latest": null,
        "size": "0.0 B"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_SNAPSHOTS_IN_HASSIO": 5,
    "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 20,
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 1.0,
    "Setting.SEND_ERROR_REPORTS": true
}
 ```
 ###### Addon Logs:
 ```
 11-01 18:09:53 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 4 seconds
11-01 18:09:57 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:09:57 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 8 seconds
11-01 18:09:58 INFO [backup.debugworker] Sending error report (see settings to disable)
11-01 18:10:05 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:10:05 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 16 seconds
11-01 18:10:21 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:10:21 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 32 seconds
11-01 18:10:53 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:10:53 ERROR [backup.model.coordinator] The addon has made too many requests to Google Drive, and will back off
11-01 18:10:53 INFO [backup.model.coordinator] I'll try again in an hour
11-01 18:12:16 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/info
11-01 18:12:16 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/info
11-01 18:12:16 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/core/info
11-01 18:12:16 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/supervisor/info
11-01 18:12:25 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/options
11-01 18:12:26 INFO [backup.model.coordinator] Syncing Snapshots
11-01 18:12:26 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
11-01 18:12:26 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1Ahyys6IuzWfGGHkQGzMFZAGFe8dKjmRX/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
11-01 18:12:26 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271Ahyys6IuzWfGGHkQGzMFZAGFe8dKjmRX%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
11-01 18:12:26 INFO [backup.drive.drivesource] Uploading '1.11.2020 OS-4.15 CORE-0.117.1 (TVJI brez clouda, rolete brez node red, ni vremena, luči delajo noč in dan, alarmi na alexi...)' to Google Drive
11-01 18:12:26 DEBUG [backup.drive.driverequests] Attempting to resume a previosuly failed upload where we left off
11-01 18:12:26 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:12:26 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 2 seconds
11-01 18:12:28 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:12:28 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 4 seconds
11-01 18:12:32 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:12:32 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 8 seconds
11-01 18:12:40 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UwqqMPpiEHhnzS-jvvwh8CDMviDqVDbz7AOCDiYUrNnurlJAKxAFuclctBWoIw3qefRJYTJ-ifRiFJDZVa-5z8mbsdGjw
11-01 18:12:40 ERROR [backup.drive.driverequests] The addon has made too many requests to Google Drive, and will back off: we'll retry in 16 seconds
 ```
 ###### Supervisor Logs:
 ```
 [32m20-11-01 17:06:47 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:06:47 INFO (MainThread) [supervisor.api.security] /snapshots/ed94214c/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:06:47 INFO (MainThread) [supervisor.api.security] /snapshots/29bf0023/info access from cebe7a76_hassio_google_drive_backup[0m
[33m20-11-01 17:07:11 WARNING (MainThread) [supervisor.api.ingress] No valid ingress session None[0m
[32m20-11-01 17:08:19 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:08:30 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:08:31 INFO (MainThread) [supervisor.api.security] /snapshots/29bf0023/download access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:08:31 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot 29bf0023[0m
[32m20-11-01 17:09:50 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:09:50 INFO (MainThread) [supervisor.api.security] /snapshots/29bf0023/download access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:09:50 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot 29bf0023[0m
[33m20-11-01 17:09:52 WARNING (MainThread) [supervisor.api.ingress] No valid ingress session None[0m
[32m20-11-01 17:10:15 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json[0m
[32m20-11-01 17:10:16 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:10:16 INFO (MainThread) [supervisor.api.security] /core/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:10:31 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/hassio-addons repository[0m
[32m20-11-01 17:10:31 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository[0m
[32m20-11-01 17:10:31 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/sabeechen/hassio-google-drive-backup repository[0m
[31m20-11-01 17:10:31 ERROR (MainThread) [supervisor.store.git] Can't update https://github.com/sabeechen/hassio-google-drive-backup repo: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch --depth=1 --update-shallow -v origin
  stderr: 'error: object file .git/objects/f7/502ec9dcfaa658ce9a7908ceada4fd61bcb0d2 is empty
fatal: loose object f7502ec9dcfaa658ce9a7908ceada4fd61bcb0d2 (stored in .git/objects/f7/502ec9dcfaa658ce9a7908ceada4fd61bcb0d2) is corrupt'.[0m
[32m20-11-01 17:10:32 INFO (MainThread) [supervisor.store] Loading add-ons from store: 67 all - 0 new - 0 remove[0m
[32m20-11-01 17:12:16 INFO (MainThread) [supervisor.api.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:12:16 INFO (MainThread) [supervisor.api.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:12:26 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:12:26 INFO (MainThread) [supervisor.api.security] /snapshots/29bf0023/download access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-01 17:12:26 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot 29bf0023[0m
[32m20-11-01 17:12:54 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
     devices = self.api.discovery()
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 188, in discovery
    response = self._request("Discovery", "discovery")
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 267, in _request
    self._raise_frequently_invoke(
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 292, in _raise_frequently_invoke
    raise TuyaFrequentlyInvokeException(message)
tuyaha.tuyaapi.TuyaFrequentlyInvokeException: Method [Discovery] fails 1 time(s) using poll interval 60.0 - error: you can discovery once in 600 seconds
2020-11-01 18:12:19 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.bfd8ea2c397343e5f6hvvo fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 278, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 474, in async_device_update
    raise exc
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/tuya/__init__.py", line 255, in update
    self._tuya.update()
  File "/usr/local/lib/python3.8/site-packages/tuyaha/devices/switch.py", line 16, in update
    return self._update(use_discovery=True)
  File "/usr/local/lib/python3.8/site-packages/tuyaha/devices/base.py", line 84, in _update
    devices = self.api.discovery()
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 188, in discovery
    response = self._request("Discovery", "discovery")
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 267, in _request
    self._raise_frequently_invoke(
  File "/usr/local/lib/python3.8/site-packages/tuyaha/tuyaapi.py", line 292, in _raise_frequently_invoke
    raise TuyaFrequentlyInvokeException(message)
tuyaha.tuyaapi.TuyaFrequentlyInvokeException: Method [Discovery] fails 2 time(s) using poll interval 60.0 - error: you can discovery once in 600 seconds

 ```
 Copy the info from the previous page here

**Comment by sabeechen** [comment]
This most likely means your Google Drive account has too many requests being made to it.  Its unlikey the addon is the cause of this problem because:

- The addon got the same error after backing off for an hour.
- This addon makes a very small number of requests compared to the maximum number that your Google Drive account can sustain.

This likely means that something else is writing to your Google Drive account and creating the problem.  What's doing that I can't speculate on, becuase it depeneds on what other things you have set up to use your Google Drive account.

Id recommend either:

- Stopping whatever else might be bogging down your Google Drive account.  Things like the [Backup and Sync](https://www.google.com/drive/download/) desktop tool might be the problem if you're using it.  You can see a list of the external applications that have access to your Google Drive account by looking at [https://myaccount.google.com/permissions](https://myaccount.google.com/permissions).
- Creating a new Google Drive account just for this addon so nothing else competes with it.

**Comment by aljazsipec** [comment]
thank you for reply.

got it working ... apearently there was trouble in the name of one of the snapshots.
after deleting it the sync went trought.
thank you.
i will bought you a coffe !!!

**Comment by sabeechen** [comment]
Strange, can you share what the name was?  If a weird name causes this error to come up, then I've got a bug somewhere.

Thanks for the coffee!

**Comment by sipeca18** [comment]
Yea sure.

1.11.2020 OS-4.15 CORE-0.117.1 (TVJI brez clouda, rolete brez node red, ni vremena, luči delajo noč in dan, alarmi na alexi...).tar

This is the name i gave for snapshot.
Hope it helps you.

**Comment by sabeechen** [comment]
Unfortunately I wasn't able to get anything strange to happen using that snapshot name on my test systems. I'm going to close this for now, but if you see the issue pop up again feel free to reopen it because I'm still interested in tracking down what was originally causing the problem.

**Comment by sabeechen** [comment]
I take that back, when I left my machine alone for a little while it came up with the exact same error as you when using that snapshot name.  This is definitely a bug.  A weird one too.

**Comment by aljazsipec** [comment]
well i hope i didnt make you to much work xD.

**Comment by sabeechen** [comment]
Looks like any file name over about 100 characters long will cause this.  I'm kind of amazed you're the first person to run into this, AFAICT tell this bug has always been there.

**Comment by sabeechen** [comment]
I've fixed this in the latest version, 0.103.0.  Please let me know if you run into any more trouble with it.
